crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -58,6 +58,10 @@ impl Rpc {
 #[derive(Clone)]
 pub struct Ethereum {
     web3: DynWeb3,
+    inner: Arc<Inner>,
```

Comment:
Why is this needed? Is there a `clone` operation you are particularly worried about (in this case should we maybe just wrap this in an Arc given we now have Arc-ception with `GasPriceEstimator`?